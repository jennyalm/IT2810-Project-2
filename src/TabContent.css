.Tab-Content{
    display: grid;
    grid-template-columns: [start] 4fr [center] 4fr [end];
    padding: 20px 0 20px 0px;
    background: white;
    height: 50vh;
    box-shadow:
            0 1px 1px rgba(0,0,0,0.15),
            0 10px 0 -5px #eee,
            0 10px 1px -4px rgba(0,0,0,0.15),
            0 20px 0 -10px #eee,
            0 20px 1px -9px rgba(0,0,0,0.15);
}


.pic{
    margin-left: 100px;
    grid-column-start: start;
    grid-column-end: center;
    height: 50%;
    width: 50%;
}

.info{
    grid-column-start: center;
    grid-column-end: end;

    display: grid;
    grid-template-rows: [top] 3fr [middle] 2fr [bottom];

    border: solid #ccc;
    border-width: 0 0 0 1px;
    padding-left: 10%;
    padding-right: 10%;
    padding-bottom: 10%;
    width: 80%;

}

.info-text{
    grid-row-start: top;
    grid-row-end: middle;
    text-align: center;
    letter-spacing: 1px;
    font-size: 130%;
    height: auto;

}

.play-button{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: auto;
    grid-row-start: middle;
    grid-row-end: bottom;

}

.tab-list {
    border-bottom: 0px solid #b7b7b7;
    padding-left: 0px;
    margin-bottom: 0;
}

.tab-list-item {
    display: inline-block;
    list-style: none;
    margin-bottom: -1px;
    padding: 0.5rem 0.75rem;
    cursor: pointer;
    width: auto;
}



.tab-list-active {
    background-color: white;
    border: solid orange;
    border-width: 0px 0px 3px 0px;
    color: #424242;

}
.play-button{
    width:100%;
}

@media only screen and (max-width: 600px){
    .Tab-Content{
        grid-template-rows: [top] 2fr [mid] 3fr [bot];
        grid-template-columns: [right] 1fr [left];
        margin: 0;
        background: white;
        max-width: 375px;
        padding: 0;
        height: auto;
        width: auto;

    }

    .play-button{
        height: 30%;
    }

    .pic{
        grid-row-start: top;
        grid-row-end: mid;
        grid-column-start: right;
        grid-column-end: left;
        margin-left: 0;
    }

    .info{
        grid-row-start: mid;
        grid-row-end: bot;
        grid-column-start: right;
        grid-column-end: left;
        border: solid #ccc;
        border-width: 1px 0 0 0px;
    }
    .info-text{
        font-size: 3vh;
    }

    .tab-list-item {
        list-style: none;
        margin-bottom: -1px;
        padding: 0.5rem 0.75rem;
        cursor: pointer;
        width: auto;


    }
    .tab-list {
        border-bottom: 0px solid #b7b7b7;
        padding-left: 0px;
        margin-bottom: 0;
    }

    .tab-item{
        width: fit-content;
    }

}


@media only screen and (min-width: 600px) and (max-width: 850px){
    .Tab-Content{
        grid-template-rows: [top] 1fr [mid] 1fr [bot];
        grid-template-columns: [right] 1fr [left];
        margin: 0;
        background: white;
        max-width: 500px;
        justify-items: center;
        padding: 0;
        height: auto;
        width:auto;
    }
    .play-button{
        height: 6vh;
    }


    .pic{
        grid-row-start: top;
        grid-row-end: mid;
        grid-column-start: right;
        grid-column-end: left;
        margin-left: 0;
        padding-right: 30px;
    }

    .info{
        grid-row-start: mid;
        grid-row-end: bot;
        grid-column-start: right;
        grid-column-end: left;
        border: solid #ccc;
        border-width: 1px 0 0 0px;
        height: 30vh;
        margin-bottom: 10%;
    }

    .info-text{
        font-size: 3vh;
    }

    .tab-list-item {
        list-style: none;
        margin-bottom: -1px;
        padding: 0.5rem 0.75rem;
        cursor: pointer;
        width: auto;
        font-size: 2vw;


    }
    .tab-list {
        border-bottom: 0px solid #b7b7b7;
        padding-left: 0px;
        margin-bottom: 0;
    }

    .tab-item{
        width: fit-content;
    }

}






