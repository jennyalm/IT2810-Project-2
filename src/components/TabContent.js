import React from "react"

import "../TabContent.css";

import PostData from "../data/posts.json"
import Tabs from "./Tabs";
import '../App.css';

import FetchImage from "./FetchImage"
import FetchText from "./FetchText"
import AudioPlayer from "./AudioPlayer"



// creates each tab from the json file in post.json, each tab consists of 3 components
// 1. FetchImage which is the gets the image
// 2. FetchText which gets the text
// 3. AudioPlayer which gets the audio file.
function TabContent(props){
    const pData = PostData.map(item =>
        <div label={item.lab} className="tab-item">
            <div className={"Tab-Content"} >
                <div className={"pic"}>
                    <FetchImage imageName={props.imageName}
                                pictureCategory={props.pictureCategory}
                    />
                </div>
                <div className={"info"}>
                    <div className={"info-text"}>
                        <FetchText
                            tabCategory={props.tabCategory}
                            textCategory={props.textCategory}
                        />
                    </div>
                    <div className={"play-button"}>
                        <AudioPlayer
                            soundCategory={props.soundCategory}
                            soundName={props.soundName}
                        />
                    </div>
                </div>
            </div>
        </div>
        )


    return(
        <Tabs handleClick={props.handleClick} tabCategory={props.tabCategory}>
            {pData}
        </Tabs>
    )
}

export default TabContent
