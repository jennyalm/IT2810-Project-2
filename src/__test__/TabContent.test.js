import React from 'react'
import renderer from 'react-test-renderer'


import TabContent from "../components/TabContent"

    test('TabContent snapshot test', () => {

    const tree = renderer.create(<TabContent />)

    expect(tree).toMatchSnapshot()

})