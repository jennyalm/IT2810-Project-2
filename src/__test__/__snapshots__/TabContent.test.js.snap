// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabContent snapshot test 1`] = `
<div
  className="tabs"
>
  <ol
    className="tab-list"
  >
    <li
      className="tab-list-item tab-list-active"
      onClick={[Function]}
    >
      kombinasjon1
    </li>
    <li
      className="tab-list-item"
      onClick={[Function]}
    >
      kombinasjon2
    </li>
    <li
      className="tab-list-item"
      onClick={[Function]}
    >
      kombinasjon3
    </li>
    <li
      className="tab-list-item"
      onClick={[Function]}
    >
      kombinasjon4
    </li>
  </ol>
  <div
    className="tab-content"
  >
    <div
      className="Tab-Content"
    >
      <div
        className="pic"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
      </div>
      <div
        className="info"
      >
        <div
          className="info-text"
        >
          <p />
        </div>
        <div
          className="play-button"
        >
          <div>
            <audio
              controls={true}
            >
              <source
                src="./Sound/undefined/undefined"
                type="audio/mpeg"
              />
            </audio>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
